### PR TITLE
d2g: log full task definition rather than just task payload

### PR DESCRIPTION
**Pull request by petemoore**
This is a follow on from [#6491](https://github.com/taskcluster/taskcluster/pull/6491#pullrequestreview-1602787612) to include full task definition when showing the output of the internal d2g conversion process.

### CODE REVIEW

**Comment by petemoore** [comment]
Note, the CI failure is independent of this change, and is addressed separately in #6522.

**Comment by petemoore** [comment]
Fixes #6521

**Review by lotas — APPROVED** [review]
👍 👍 👍

File: workers/generic-worker/payload_linux.go [hunk 1]
```diff
@@ -30,23 +33,48 @@ func (task *TaskRun) convertDockerWorkerPayload() *CommandExecutionError {
...
+
+	d2gConvertedTaskDefinitionYAML, err := yaml.JSONToYAML(d2gConvertedTaskDefinitionJSON)
+	if err != nil {
+		return executionError(internalError, errored, fmt.Errorf("Could not convert task definition from JSON to YAML: %v\n", err))
+	}
+
+	task.Warn("This task was designed to run under Docker Worker. Docker Worker is a worker implementation")
+	task.Warn("that is _no longer_ maintained.")
+	task.Warn("In order to execute this task, it is being converted to a Generic Worker task, using the D2G")
+	task.Warn("utility (Docker worker 2 Generic worker):")
```
Comment (matt-boris) [inline workers/generic-worker/payload_linux.go:68]:
```suggestion
	task.Warn("utility (Docker Worker 2 Generic Worker):")
```

**Review by matt-boris — COMMENTED** [review]
Changes look great Pete, thank you! I left one quick suggestion for consistency across the message to the user.

**Pull request merged**